### PR TITLE
Port Ikfast kinematics solver

### PR DESCRIPTION
**Pull request by JafarAbdi**
### Description

eigen_conversion isn't ported yet to ROS2 so I copied the needed function and dropped its dependency

I tested it with panda robot steps are as follow [Source](https://ros-planning.github.io/moveit_tutorials/doc/ikfast/ikfast_tutorial.html)
1- clone https://github.com/JafarAbdi/panda_moveit_config/tree/ros2
2- cd moveit_resources/panda_description/urdf
3- ros2 run moveit_kinematics auto_create_ikfast_moveit_plugin.sh panda.urdf panda_arm panda_link0 panda_link8
4- now move the generated package to the main source path and recompile (the generated package name is panda_panda_arm_ikfast_plugin)

### Checklist
- [ ] **Required by CI**: Code is auto formatted using [clang-format](http://moveit.ros.org/documentation/contributing/code)
- [ ] Extend the tutorials / documentation [reference](http://moveit.ros.org/documentation/contributing/)
- [ ] Document API changes relevant to the user in the MIGRATION.md notes
- [ ] Create tests, which fail without this PR [reference](https://ros-planning.github.io/moveit_tutorials/)
- [ ] Include a screenshot if changing a GUI
- [ ] While waiting for someone to review your request, please help review [another open pull request](https://github.com/ros-planning/moveit/pulls) to support the maintainers

[//]: # "You can expect a response from a maintainer within 7 days. If you haven't heard anything by then, feel free to ping the thread. Thank you!"

### CODE REVIEW

File: moveit_kinematics/ikfast_kinematics_plugin/templates/ikfast61_moveit_plugin_template.cpp [hunk 1]
```diff
@@ -41,13 +41,30 @@
...
+template <typename T>
+void transformEigenToKDLImpl(const T& e, KDL::Frame& k)
+{
+  for (unsigned int i = 0; i < 3; ++i)
+    k.p[i] = e(i, 3);
+  for (unsigned int i = 0; i < 9; ++i)
+    k.M.data[i] = e(i / 3, i % 3);
+}
+
+void transformEigenToKDL(const Eigen::Isometry3d& e, KDL::Frame& k)
```
Comment (henningkayser) [inline moveit_kinematics/ikfast_kinematics_plugin/templates/ikfast61_moveit_plugin_template.cpp:63]: Do we need this template specification? I would remove it and simply call the template `transformEigenToKDL`

**Comment by henningkayser** [comment]
Also, please enable https://github.com/ros-planning/moveit2/blob/master/.travis.yml#L31 if possible

File: .travis.yml [hunk 2]
```diff
@@ -28,8 +28,7 @@ jobs:
     - compiler: clang  # test_ikfast_plugins takes ~10 minutes: include here to keep the main jobs shorter
       env: TEST=clang-tidy-fix
            CXXFLAGS="-Wall -Wextra -Wwrite-strings -Wunreachable-code -Wpointer-arith -Wredundant-decls -Wno-overloaded-virtual"
-           # TODO(mlautman): Enable once franka_ikfast_plugin and panda_ikfast_plugin are available
-           # BEFORE_DOCKER_SCRIPT="source moveit_kinematics/test/test_ikfast_plugins.sh"
+           BEFORE_DOCKER_SCRIPT="source moveit_kinematics/test/test_ikfast_plugins.sh"
```
Comment (henningkayser) [inline .travis.yml:31]: nice

**Comment by JafarAbdi** [comment]
> Great work! The IKFast test seems to be missing some deps, though https://travis-ci.com/github/ros-planning/moveit2/jobs/349058239#L239.

Thanks for catching it! I found out that the same happens for the main MoveIt repo [see](https://travis-ci.com/github/ros-planning/moveit/jobs/343179556#L229), the dependency installation got removed in this PR https://github.com/ros-planning/moveit/pull/1492/files

File: moveit_kinematics/test/test_ikfast_plugins.sh [hunk 3]
```diff
@@ -8,10 +8,14 @@
 # using the script auto_create_ikfast_moveit_plugin.sh
 
 # Clone moveit_resources for URDFs. They are not available before running docker.
-travis_run git clone -q --depth=1 https://github.com/ros-planning/moveit_resources /tmp/resources
+travis_run git clone -q --depth=1 -b ros2 https://github.com/ros-planning/moveit_resources /tmp/resources
 fanuc=/tmp/resources/fanuc_description/urdf/fanuc.urdf
 panda=/tmp/resources/panda_description/urdf/panda.urdf
 
+# Install lxml required for create_ikfast_moveit_plugin.py
```
Comment (henningkayser) [inline moveit_kinematics/test/test_ikfast_plugins.sh:15]: Shouldn't we just add `python-lxml` as <test_depend/>?

Comment (henningkayser) [reply moveit_kinematics/test/test_ikfast_plugins.sh:15]: Oh, this won't work since it's run before anything else...

**Pull request merged**